crates/forge/src/compatibility_check.rs

```diff
@@ -115,10 +117,80 @@ fn get_raw_version(name: &str, command: &RefCell<Command>) -> Result<String> {
         .to_string())
 }
 
+fn verify_contracts_sierra_versions(forge_config: &ForgeConfig) -> Result<()> {
+    if let Some(contract_names) = forge_config
+        .test_runner_config
+        .contracts_data
+        .get_contract_names()
+    {
+        for contract in contract_names {
+            let sierra_str = &forge_config
+                .test_runner_config
+                .contracts_data
+                .get_artifacts(contract)
+                .context(format!("Missing Sierra JSON for contract: {contract}"))?
+                .sierra;
+
+            let contract_version = parse_sierra_version(sierra_str)?;
+            if MINIMAL_SIERRA_VERSION_FOR_SIERRA_GAS > contract_version {
+                bail!(
+                    "Contract version {} is lower than required minimal sierra version",
```

Comment:
Can we do something along the lines of this?

```suggestion
                    "Contract version {} is lower than required minimal sierra version {MINIMAL_SIERRA_VERSION_FOR_SIERRA_GAS}",
```